cpp/src/column/column_device_view.cu

```diff
@@ -116,7 +113,33 @@ column_device_view::create(column_view source, cudaStream_t stream) {
 
   CUDA_TRY(cudaStreamSynchronize(stream));
 
-  return p;
+  return result;
+}
+
+} // namespace
+
+// Place any child objects in host memory (h_ptr) and use the device
+// memory ptr (d_ptr) to set any child object pointers.
+column_device_view::column_device_view( column_view source, void * h_ptr, void* d_ptr )
+    : detail::column_device_view_base{source.type(),       source.size(),
+                                      source.head(),       source.null_mask(),
+                                      source.null_count(), source.offset()},
+      _num_children{source.num_children()}
+{
```

Comment:
I feel like this constructor (and the mutable_column_view equivalent) should be made private (and child_columns_to_device_array / create_device_view_from_view made private class members.)

This feels like a detail external users should never even be thinking about.

Reply:
This function is also called by `table_device_view` to create similar children array of device views so it cannot be made private.